stdlib/public/runtime/Private.h

```diff
@@ -43,15 +43,10 @@ class TypeReferenceOwnership {
 public:
   constexpr TypeReferenceOwnership() : Data(0) {}
 
-  bool isWeak() const { return Data & Weak; }
-  bool isUnowned() const { return Data & Unowned; }
-  bool isUnmanaged() const { return Data & Unmanaged; }
-
-  void setWeak() { Data |= Weak; }
-
-  void setUnowned() { Data |= Unowned; }
-
-  void setUnmanaged() { Data |= Unmanaged; }
+#define REF_STORAGE(Name, ...) \
+  void set##Name() { Data |= Name; } \
+  bool is##Name() const { return Data & Name; }
+#include "swift/AST/ReferenceStorage.def"
```

Comment:
Do you want to metaprogram the actual flags, too?  That's okay if this isn't ABI-exposed.

Reply:
I don't think there's actually a reason for this to be using a manual bitmask anyway instead of bit-fields.

Reply:
The assignment of a new reference storage type to a flag is not meta-programmed, but the fallout from adding a new flag is. In other words, the build will fail until a programmer manually updates the enum within `TypeReferenceOwnership` at the top of stdlib/public/runtime/Private.h. I can't see a problem with this, can you?